code/parse/sexp.cpp

```diff
@@ -2636,13 +2636,20 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, s
 						if (get_operator_const(op_node) < First_available_operator_id) {
 							ship_node = CDR(op_node);
 						} else {
-							ship_node = get_dynamic_parameter_index(Sexp_nodes[op_node].text, argnum);
-
-							if (ship_node < 0)
-								error_display(1, "Expected to find a dynamic lua parent parameter for node %i in operator %s but found nothing!",
+							int r_count = get_dynamic_parameter_index(Sexp_nodes[op_node].text, argnum) + 1; //account for the node header
+							
+							if (r_count < 0)
```

Comment:
The +1 in the previous line is going to prevent `(r_count < 0)` from ever being true.  You could alternatively adjust the `while` condition in the subsequent loop, or turn it into a `for` loop.

Reply:
Done

Reply:
done again